app/src/main/java/com/cornellappdev/uplift/ui/components/gymdetail/GymTodaysClasses.kt

```diff
@@ -1,26 +1,35 @@
 package com.cornellappdev.uplift.ui.components
 
+import androidx.compose.foundation.background
 import androidx.compose.foundation.layout.*
 import androidx.compose.material.Text
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.text.style.TextAlign
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
+import androidx.navigation.NavHostController
 import com.cornellappdev.uplift.models.Gym
+import com.cornellappdev.uplift.ui.viewmodels.ClassDetailViewModel
 import com.cornellappdev.uplift.util.montserratFamily
 
 /**
  * A vertical list of classes that a gym offers today. Refers to [ClassInfoCard] to create the class
  * cards.
  */
 @Composable
-fun GymTodaysClasses(gym: Gym) {
+fun GymTodaysClasses(
+    gym: Gym,
+    classDetailViewModel: ClassDetailViewModel,
```

Comment:
Why was the reason you changed back to passing it in instead of using the viewModel function?

Reply:
the `viewModel()` function doesn't seem to work when it's called in different parts of the codebase. I'm not entirely sure why, but basically if `viewModel()` is called on at some time, perhaps on the main thread, and then `viewModel()`is called again later on some other thread or at some other place in the codebase (I'm not entirely sure), it actually ends up returning a different instance. Due to its unreliability, I've switched back to passing them in.